### PR TITLE
Feature/issue 2449 algebra solver

### PR DESCRIPTION
**Pull request by charlesm93**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py src/test/unit`
- [ ] Run cpplint: `make cpplint` (had an issue when trying to run it -- maybe an issue on my local machine)
- [x] Declare copyright holder and open-source license: see below

#### Summary
Solves issue # 2449. Update unit tests to test cases where parameter vector (`theta`) has length 0 and initial guess is passed as a parameter vector. Update manual to reflect removed constraints.

#### How to Verify
unit tests

#### Documentation
Stan user manual.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Charles Margossian (Columbia University)

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by charlesm93** [comment]
I'll add to the manual the algebraic solver only uses first-order derivatives (as oppose to using second-order derivatives as you might with Newton's method).

**Comment by charlesm93** [comment]
Could someone help me interpret Jenkins' error message?

`assigning to 'double' from incompatible type 'const stan::math::var'` -- fair enough. But the unit tests work, and there is no indication about which file specifically might be failing.

**Comment by bob-carpenter** [comment]
If you scroll down to the botto of the console output, there's a problem with the integration tests.  It's exactly as you say that something there is trying to assign `var` to a `double`.  The integration tests will test that all the models can be compiled, which will test various instantiations.

**Comment by charlesm93** [comment]
*smack on the forehead!!* It's because the math pull request isn't in yet! The tests pass in a parameter as the initial guess, which is not allowed in the current version of the algebraic solver.

**Comment by charlesm93** [comment]
Ok, merged the pull request in math!

**Comment by bob-carpenter** [comment]
jenkins, retest this please.

File: src/docs/stan-reference/examples.tex [hunk 1]
```diff
@@ -8174,7 +8175,7 @@ \subsection{Pathological Solutions}
 algebraic solver will not report these cases, as the algorithm stops once it has found
 an acceptable solution. The initial guess will often determine which solution gets found
 first. The degeneracy may be broken by putting additional constraints on the solution.
-For instance, it might make physical sense for a solution to be positive or negative.
+For instance, it might make ``physical sense" for a solution to be positive or negative.
```
Comment (bob-carpenter) [inline src/docs/stan-reference/examples.tex]: This needs to be ` ``physical sense'' ` with two apostrophes trailing to render correctly in LaTeX.

File: src/docs/stan-reference/functions.tex [hunk 2]
```diff
@@ -3693,7 +3693,7 @@ \subsection{Arguments to the Algebraic Solver}
   (4) integer data, and the return value contains the value of the algebraic
   function, which goes to 0 when we plug in the solution to the algebraic system,
 \item \farg{y\_guess}: initia guess for the solution, type \code{vector},
-\item \farg{theta}: parameters, type \code{vector},
+\item \farg{theta}: parameters, type \code{vector}, parameters only,
```
Comment (bob-carpenter) [inline src/docs/stan-reference/functions.tex]: I think this is contradictory---it should be "parameters" or "parameters only"

**Pull request merged**